indexing-service/src/main/java/org/apache/druid/indexing/worker/executor/ExecutorLifecycle.java

```diff
@@ -80,7 +83,7 @@ public ExecutorLifecycle(
     this.taskConfig = taskConfig;
     this.taskActionClientFactory = taskActionClientFactory;
     this.taskRunner = taskRunner;
-    this.jsonMapper = jsonMapper;
+    this.jsonMapper = new DefaultObjectMapper((DefaultObjectMapper) jsonMapper).addMixIn(PasswordProvider.class, PasswordProviderRedactionMixIn.class);
```

Comment:
Would you please explain why this change is needed? `KinesisIndexTask` is already using `PasswordProvider`, so it should work without this change.

Reply:
@jihoonson When I do not swap out the ObjectMapper with a new one that has the MixIn, the password is written in plaintext to the metastore and the task log file. Both of those instances serialize the object to JSON using the existing ObjectMapper before writing, so I don't think that the ObjectMapper is configured to recognize and omit the password as needed. Also, since the way the indexing service works requires the Serialization to selectively include the password in the json (when the overlord is assigning a remote worker), changing the default object mapper to handle this new block would not work either. That is why I felt the need to use a MixIn

Reply:
I see. It sounds good to me. Would you please this to the javadoc of `PasswordProviderRedactionMixIn`?

Reply:
@jihoonson Sorry, but I'm unclear about what change you are requesting in the javadoc. Could you elaborate?

Reply:
Ah I meant it would be nice if you can add javadoc for `PasswordProviderRedactionMixIn` explaining why that class exists and how it's being used.

Reply:
Sure, I added a Javadoc. Not sure if there is an expected pattern for this Project. I wasn't able to find too much consistency across the project. Will be happy to get in line if there is an expected standard out there